### DISCUSSION TITLE
Multicore and EEPROM

### DISCUSSION THREAD

**Discussion by LazaroFilm** [earlephilhower/arduino-pico#1757] [General]
I'm working on a project where I am trying to have an IMU sensor and Fusion algorithm running on `core1` while `core0` controls everything else (TFT screen, Touch, LVGL, Neopixels, and settings saves via EEPROM). 

I'm using the [multicore](https://arduino-pico.readthedocs.io/en/latest/multicore.html) because I want the IMU and Fusion to run in a loop as fast as possible independently from what `core0` does (the screen refreshing graphics can slow down the loop pretty badly). I am running into the situation where saving EEPROM freezes the whole program sometimes. 

I tried using `rp2040.idleOtherCore()` but I'm not sure I'm using it properly. Can someone give me a simple exemple of a loop running on `core1` while saving a value to EEPROM on `core0`?

Also, I'm saving the values from core1 to global variable defined at the top of the sketch and it seems to work. Is that okay, or should I use the FIFO (Which I don't quite understand either)

**Comment by earlephilhower** [comment]
> ...I tried using rp2040.idleOtherCore() but...

There's almost no reason in the world to use this outside of writing to flash memory outside of the `EEPROM` and `LittleFS` libraries (i.e. using the SDK calls directly).  If you want atomic access to a multi-word state use an SDK `mutex` or semaphore.  If it's just a word(32b) long element you need to protect, don't bother because 32b writes are atomic.  (16b and 8b writes are not, however, AFAIK).

Writing to EEPROM *has* to freeze the other core because when you write to flash, you can't read from it.  So the non-writing core needs to jump to an code-in-RAM busy loop while the writing is going on (hence, the embedded `idleOtherCore` call).  It does this automatically, there is no need for you to do anything yourself other than use the `EEPROM` class.
